tests/integration/conftest.py

```diff
@@ -4,136 +4,73 @@
 Copyright (c) 2019 Aiven Ltd
 See LICENSE for details
 """
+from _pytest.fixtures import SubRequest
+from aiohttp.pytest_plugin import AiohttpClient
+from aiohttp.test_utils import TestClient
 from contextlib import closing, ExitStack
-from dataclasses import asdict, dataclass
+from dataclasses import asdict
 from filelock import FileLock
 from kafka import KafkaProducer
-from kafka.errors import LeaderNotAvailableError, NoBrokersAvailable
 from karapace.config import set_config_defaults, write_config
 from karapace.kafka_rest_apis import KafkaRest, KafkaRestAdminClient
 from karapace.schema_registry_apis import KarapaceSchemaRegistry
 from karapace.utils import Client
 from pathlib import Path
 from subprocess import Popen
-from tests.utils import (
-    Expiration,
-    get_random_port,
-    KAFKA_PORT_RANGE,
-    KafkaConfig,
+from tests.integration.utils.config import KafkaConfig, KafkaDescription, ZKConfig
+from tests.integration.utils.kafka_server import (
+    configure_and_start_kafka,
     KafkaServers,
-    new_random_name,
-    REGISTRY_PORT_RANGE,
-    repeat_until_successful_request,
-    ZK_PORT_RANGE,
+    maybe_download_kafka,
+    wait_for_kafka,
 )
-from typing import AsyncIterator, Dict, Iterator, List, Optional, Tuple
-
+from tests.integration.utils.network import get_random_port, KAFKA_PORT_RANGE, REGISTRY_PORT_RANGE, ZK_PORT_RANGE
+from tests.integration.utils.process import stop_process, wait_for_port_subprocess
+from tests.integration.utils.synchronization import lock_path_for
+from tests.integration.utils.zookeeper import configure_and_start_zk
+from tests.utils import new_random_name, repeat_until_successful_request
+from typing import AsyncIterator, Iterator, Optional, Tuple
+
+import asyncio
 import os
+import pathlib
 import pytest
-import signal
-import socket
 import time
 import ujson
 
-# Keep these in sync with the Makefile
-KAFKA_CURRENT_VERSION = "2.7"
-BASEDIR = "kafka_2.13-2.7.0"
-
-CLASSPATH = os.path.join(BASEDIR, "libs", "*")
+REPOSITORY_DIR = pathlib.Path(__file__).parent.parent.parent.absolute()
+RUNTIME_DIR = REPOSITORY_DIR / "runtime"
+TEST_INTEGRATION_DIR = REPOSITORY_DIR / "tests" / "integration"
 KAFKA_WAIT_TIMEOUT = 60
-
-
-@dataclass
-class ZKConfig:
```

Comment:
I moved almost everything that is not a fixture into the `tests/integration/utils` module.